src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnusedImportsTest.java

```diff
@@ -0,0 +1,81 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck;
+
+public class XpathRegressionUnusedImportsTest extends AbstractXpathTestSupport {
+
+    private final String checkName = UnusedImportsCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testOne() throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionUnusedImportsOne.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(UnusedImportsCheck.class);
+
+        final String[] expectedViolation = {
+            "3:8: " + getCheckMessage(UnusedImportsCheck.class,
+                    UnusedImportsCheck.MSG_KEY, "java.util.List"),
+        };
+
+        final List<String> expectedXpathQueries = Collections.singletonList(
+                "/IMPORT/DOT[./IDENT[@text='List']]/DOT/IDENT[@text='java']");
```

Comment:
This will suppress any import that matches `import java.*.List`, i.e.
```
import java.otherpackage.List;
```
I'm not sure they ever exist, so it's OK. However, the bad thing is that users can use this xpath as an example.

Reply:
@pbludov I believe this regression should be checking the output from the CLI when verifying the xpath. If you feel it is being too loose, then please make an issue on it.
https://github.com/checkstyle/checkstyle/blob/b5ab579539b110c7ddf59f627767d2188f4a869a/src/it/java/org/checkstyle/suppressionxpathfilter/AbstractXpathTestSupport.java#L67